### ISSUE TITLE
Akka Remoting hangs on empty connections instead of timing out

### ISSUE THREAD

**Issue by drewhk** [akka/akka#19748]
Simple reproducer:

```scala
val remote = Tcp().outgoingConnection("127.0.0.1", 2552)
Source.maybe[ByteString].via(remote).log("FUZZER: ").to(Sink.ignore).run()
```

The connections just stays around in limbo without timing out.

While this one:

```scala
(Source.single(ByteString(0, 0, 0, 1, 0)) ++ Source.maybe).via(remote).log("FUZZER ").to(Sink.ignore).run
```

fails properly with:

```
akka.remote.transport.AkkaProtocolException: Error while decoding incoming Akka PDU of length: 1
Caused by: akka.remote.transport.PduCodecException: Decoding PDU failed.
        at akka.remote.transport.AkkaPduProtobufCodec$.decodePdu(AkkaPduCodec.scala:167)
        at akka.remote.transport.ProtocolStateActor.akka$remote$transport$ProtocolStateActor$$decodePdu(AkkaProtocolTransport.scala:580)
        at akka.remote.transport.ProtocolStateActor$$anonfun$3.applyOrElse(AkkaProtocolTransport.scala:375)
        at akka.remote.transport.ProtocolStateActor$$anonfun$3.applyOrElse(AkkaProtocolTransport.scala:343)
        at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
        at akka.actor.FSM$class.processEvent(FSM.scala:604)
        at akka.remote.transport.ProtocolStateActor.processEvent(AkkaProtocolTransport.scala:269)
        at akka.actor.FSM$class.akka$actor$FSM$$processMsg(FSM.scala:598)
        at akka.actor.FSM$$anonfun$receive$1.applyOrElse(FSM.scala:592)
        at akka.actor.Actor$class.aroundReceive(Actor.scala:467)
        at akka.remote.transport.ProtocolStateActor.aroundReceive(AkkaProtocolTransport.scala:269)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516)
        at akka.actor.ActorCell.invoke(ActorCell.scala:487)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:238)
        at akka.dispatch.Mailbox.run(Mailbox.scala:220)
```

**Comment by drewhk** [comment]
Also, let's add akka-stream as a dependecy to remote-tests. The tooling it provides is indispensable. Unfortunately the side-effect will be that PR validation would run remote-tests on streams changes :(

**Comment by drewhk** [comment]
Seems like already fixed in 2.4.x: https://github.com/akka/akka/pull/18542

This means that this is a backport to 2.3.x
